cmd/wcc/main.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/cmd/bnsd/app"
+	"github.com/iov-one/weave/cmd/bnsd/client"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/crypto"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/iov-one/weave/x/gov"
+)
+
+var commands = map[string]func(input io.Reader, output io.Writer, args []string) error{
+	"transfer-proposal": cmdNewTransferProposal,
+	"sign":              cmdSignTransaction,
+	"submit":            cmdSubmitTx,
+}
+
+func main() {
+	if len(os.Args) == 1 {
+		available := make([]string, 0, len(commands))
+		for name := range commands {
+			available = append(available, name)
+		}
+		fmt.Fprintf(os.Stderr, "Usage: %s <cmd> [<flags>]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "\nAvailable commands are:\n\t%s\n", strings.Join(available, "\n\t"))
+		os.Exit(2)
+	}
+	run, ok := commands[os.Args[1]]
+	if !ok {
+		available := make([]string, 0, len(commands))
+		for name := range commands {
+			available = append(available, name)
+		}
+		fmt.Fprintf(os.Stderr, "Unknown command %q\n", os.Args[1])
+		fmt.Fprintf(os.Stderr, "\nAvailable commands are:\n\t%s\n", strings.Join(available, "\n\t"))
+		os.Exit(2)
+	}
+
+	// Skip two first as second argument is the command name that we just
+	// consumed.
+	if err := run(os.Stdin, os.Stdout, os.Args[2:]); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func cmdNewTransferProposal(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprintln(flag.CommandLine.Output(), `
+Transfer funds from source account to the destination.
+		`)
+		fl.PrintDefaults()
+	}
+	var (
+		srcFl    = flAddress(fl, "src", "", "A source account address that the founds are send from.")
+		dstFl    = flAddress(fl, "dst", "", "A destination account address that the founds are send to.")
+		amountFl = flCoin(fl, "amount", "1 IOV", "An amount that is to be transferred between the source to the destination accounts.")
+		memoFl   = fl.String("memo", "", "A short message attached to the transfer operation.")
+		titleFl  = fl.String("title", "Transfer funds to distribution account", "The proposal title.")
+		descFl   = fl.String("description", "Transfer funds to distribution account", "The proposal description.")
+		eRuleFl  = fl.String("electionrule", "", "The ID of the election rule to be used.")
+	)
+	fl.Parse(args)
+
+	option := app.ProposalOptions{
+		Option: &app.ProposalOptions_SendMsg{
```

Comment:
What I understood from https://github.com/iov-one/weave/issues/626#issuecomment-492641972 is that there are 2 payload types that we need to support for each of the two steps.
1. [Release from escrow](https://github.com/iov-one/weave/blob/master/cmd/bnsd/app/codec.proto#L105 )
2. Cash transfer

Release from escrow is not supported

Reply:
Good catch! :+1: